test/test_linalg.py

```diff
@@ -2696,6 +2697,59 @@ def check(equation, *operands):
         # with strided tensors
         check("bn,anm,bm->ba", l[:, ::2], w[:, ::2, ::2], r[:, ::2])
 
+    @dtypes(torch.double)
```

Comment:
I added a random test to hopefully catch anything missed in the manual test cases.